jina/peapods/pods/__init__.py

```diff
@@ -31,7 +31,12 @@ def __init__(self, args: 'argparse.Namespace', needs: Set[str] = None):
         self.deducted_head = None
         self.deducted_tail = None
 
-        self.peas_args = self._parse_args(args)
+        if isinstance(args, Dict):
+            # This is used when a Pod is created in a remote context, where peas & their connections are already given.
+            self.peas_args = args
+        else:
+            self.peas_args = self._parse_args(args)
```

Comment:
when is dict used? inside `JinadRuntime` sending this to `jinad`?

Reply:
https://github.com/jina-ai/jina/blob/master/jina/peapods/runtimes/jinad/__init__.py#L28

before sending these args, they're converted to dict & on the receiving end, jinad converts dict to `argparse`